<?xml version="1.0"?>
<!-- G100 = 8036, FOV = 100 -->
<launch>
  <arg name="pi"   value="3.1415926535897932" />
  <arg name="pi/2" value="1.5707963267948966" />
  <arg name="base_frame" value="dm_base_frame" />
  
  <arg name="type_color_only"           default="0" />
  <arg name="type_8_bits"               default="1" />
  <arg name="type_14_bits"              default="2" />
  <arg name="type_11_bits"              default="4" />


  <!-- please set the related parameters based on the module information for single module -->
  <arg name="camera" value="8073" />
  <arg name="tf_prefix" value="$(arg camera)" />

    <include file="$(find dm_preview)/launch/dm_preview.launch">
     <arg name="tf_prefix" value="$(arg camera)" />
     <arg name="module_name" value="$(arg camera)" />
     <arg name="dev_serial_number" value="8073051Z20015" />
     
     <!-- color -->
     <arg name="color_width" value="1104" />
     <arg name="color_height" value="848" />
     <arg name="depth_width" value="1104" />
     <arg name="depth_height" value="848" />
     <arg name="framerate" default="30" />
     <arg name="depth_data_type" default="$(arg type_11_bits)" />
     
     <arg name="depth_maximum_mm" value="4000" />
     <arg name="ir_intensity" default="60" />

    </include>

    <node pkg="tf" type="static_transform_publisher" name="point_cloud_8036_broadcaster"
        args="0 0.03 0 0 0 0 $(arg base_frame) $(arg tf_prefix)_points_frame 30" />
  

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find dm_preview)/rviz/8036.rviz" output="screen"/>
</launch>
